app/controllers/application_controller.rb

```diff
@@ -35,10 +35,17 @@ def delete_otp_session_variables
   private
 
   def set_organization
-    @organization = if params[:organization_id].present?
-                      Organization.find(params[:organization_id])
-                    else
-                      Organization.singleton
-                    end
+    organization = if params[:organization_id].present?
```

Comment:
just as an idea for a different possible implementation:
```
def set_organization
   collection = current_user.admin? ? Organization : current_user.organizations
   collection.find(params[:organization_id]) if params[:organization_id].present?
   ...
```

the additional user_permitted? method would then be obsolete. What I like on your approach is that is quite explicit on checking but it has some overhead compared to the above.